File: src/main/javaHexagonal/com/template/domain/repositories/UserRepository.java
package com.template.domain.repositories;

import java.util.Optional;
import com.template.domain.models.User;
import java.util.List;

public interface UserRepository {
	Optional<User> findById(Long id);

	User save(User user);

	List<User> findAll();

	void deleteById(Long id);
}

File: src/main/kotlinHexagonal/com/template/domain/repositories/UserRepository.kt
package com.template.domain.repositories

import java.util.Optional
import com.template.domain.models.User
import kotlin.collections.MutableList

interface UserRepository {
    fun findById(id: Long): User?
    fun save(user: User): User?
    fun findAll(): MutableList<User?>
    fun deleteById(id: Long)
}